api/src/main/java/io/opentelemetry/api/trace/Span.java

```diff
@@ -304,17 +304,18 @@ default Span setAttribute(AttributeKey<Long> key, int value) {
   void end();
 
   /**
-   * Marks the end of {@code Span} execution with the specified {@link EndSpanOptions}.
+   * Marks the end of {@code Span} execution with the specified timestamp.
    *
    * <p>Only the timing of the first end call for a given {@code Span} will be recorded, and
    * implementations are free to ignore all further calls.
    *
    * <p>Use this method for specifying explicit end options, such as end {@code Timestamp}. When no
    * explicit values are required, use {@link #end()}.
    *
-   * @param endOptions the explicit {@link EndSpanOptions} for this {@code Span}.
+   * @param timestamp the explicit timestamp, as nanos from the epoch, for this {@code Span}. {@code
+   *     0} indicates current time should be used.
```

Comment:
As usual, I'm not a fan of using `0` as a magic number here. Would prefer something that was obviously not a valid timestamp, Long.MIN_VALUE or something.

Reply:
does this method need any magic for current time, since `end()` supports that?

Reply:
I don't think so either but that's the same logic as we have right now - can we update the semantics separately? Want to get the API locked in

Reply:
I wonder if no one noticed if I hadn't updated javadoc :-)

Reply:
I did see the zeros floating around in tests and such and it was starting to make me grumpy, even before I saw the javadoc. :)

Reply:
Before this PR, zero was an undocumented magic number. I guess it's better to have it documented, if we're going to use it! But, I'd rather not have the magic '0' at all, personally. I mean, maybe you want a span to end at unix epoch! (ok, not really)